pkg/ebpftracer/decoder/decoder.go

```diff
@@ -675,31 +675,37 @@ func (decoder *Decoder) ReadProtoDNS() (*types.ProtoDNS, error) {
 		return nil, err
 	}
 
-	payload, subProtocol, err := packet.ExtractPayload(data)
+	details, err := packet.ExtractPacketDetails(data)
 	if err != nil {
 		return nil, err
 	}
 
-	if subProtocol == packet.SubProtocolTCP {
-		if len(payload) < 2 {
+	if details.Proto == packet.SubProtocolTCP {
+		if len(details.Payload) < 2 {
 			return nil, errDNSMessageNotComplete
 		}
 
 		// DNS over TCP prefixes the DNS message with a two octet length field. If the payload is not as big as this specified length,
 		// then we cannot parse the packet, as part of the DNS message will be send in a later one.
 		// For more information see https://datatracker.ietf.org/doc/html/rfc1035.html#section-4.2.2
-		length := int(binary.BigEndian.Uint16(payload[:2]))
-		if len(payload)+2 < length {
+		length := int(binary.BigEndian.Uint16(details.Payload[:2]))
+		if len(details.Payload)+2 < length {
 			return nil, errDNSMessageNotComplete
 		}
-		payload = payload[2:]
+		details.Payload = details.Payload[2:]
 	}
-	if err := dnsPacketParser.DecodeFromBytes(payload, gopacket.NilDecodeFeedback); err != nil {
+	if err := dnsPacketParser.DecodeFromBytes(details.Payload, gopacket.NilDecodeFeedback); err != nil {
 		return nil, err
 	}
 
 	pbDNS := &castpb.DNS{
 		Answers: make([]*castpb.DNSAnswers, len(dnsPacketParser.Answers)),
+		Tuple: &castpb.Tuple{
+			SrcIp:   details.Src.Addr().AsSlice(),
+			DstIp:   details.Dst.Addr().AsSlice(),
+			SrcPort: uint32(details.Src.Port()),
```

Comment:
Is there a reason why we're using uint32 instead of uint16 (like in the kernel) for ports?

Reply:
Sadly in protobuf there is no `uint16`, hence we use `uint32`.